### PR TITLE
Some improvements to ES812PostingsReader

### PR DESCRIPTION
**Pull request by original-brownbear**
Removing some fields that need not exist from this class as well as cleaning up two redundant conditions and adding missing `final`. Mainly noticed this because these things are taking up quite a bit of heap, though the savings here are relatively minor they're still measurable and changes make the code simpler outright as well.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/codec/postings/ES812PostingsReader.java [hunk 1]
```diff
@@ -265,9 +265,7 @@ public ImpactsEnum impacts(FieldInfo fieldInfo, BlockTermState state, int flags)
             return new BlockImpactsDocsEnum(fieldInfo, (IntBlockTermState) state);
         }
 
-        if (indexHasPositions
-            && PostingsEnum.featureRequested(flags, PostingsEnum.POSITIONS)
-            && (indexHasOffsets == false || PostingsEnum.featureRequested(flags, PostingsEnum.OFFSETS) == false)
+        if ((indexHasOffsets == false || PostingsEnum.featureRequested(flags, PostingsEnum.OFFSETS) == false)
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/index/codec/postings/ES812PostingsReader.java:268]: This is covered by the previous condition above.

File: server/src/main/java/org/elasticsearch/index/codec/postings/ES812PostingsReader.java [hunk 2]
```diff
@@ -287,8 +285,6 @@ final class BlockDocsEnum extends PostingsEnum {
         private ES812SkipReader skipper;
         private boolean skipped;
 
-        final IndexInput startDocIn;
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/index/codec/postings/ES812PostingsReader.java:290]: The input is a final field on the outer class, no need to keep a separate reference to that index input in a non-static inner class.

File: server/src/main/java/org/elasticsearch/index/codec/postings/ES812PostingsReader.java [hunk 3]
```diff
@@ -1527,20 +1513,10 @@ final class BlockImpactsEverythingEnum extends ImpactsEnum {
         // before reading payloads/offsets:
         private long payPendingFP;
 
-        // Where this term's postings start in the .doc file:
-        private long docTermStartFP;
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/index/codec/postings/ES812PostingsReader.java:1531]: Same here these can be inlined away.

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
@elasticmachine update branch

**Comment by original-brownbear** [comment]
Thanks Martijn!

**Pull request merged**